### PR TITLE
feat: Add management API and CLI for listing partitions

### PR DESCRIPTION
**Pull request by alamb**
Re https://github.com/influxdata/influxdb_iox/issues/942 and https://github.com/influxdata/influxdb_iox/issues/979

# Rationale:
I need a place to hang the CLI commands to move chunks in a partition (https://github.com/influxdata/influxdb_iox/issues/979)

# Changes:
* Add RPC definition + implementations for listing partitions
* Add new cli command `influxdb_iox database partition list db_name` to get a list of partition keys

# Follow on work
There is a bunch of information about a partition that should probably be included in the output of `database partition get` but I don't need it for my demo.

Thus, I figured I would get what I did need (`list`) up alongside the basic plumbing for `get` and we can fill in the details later

# Examples
List all partition keys;
```
./influxdb_iox database partition list my_db
[
  "2020-06-11 16:00:00",
  "2021-03-12 14:00:00"
]
```

Get detail of a particular partiton key (nothing interesting yet):
```
./influxdb_iox database partition get my_db "2021-03-12 14:00:00"
{
  "key": "2021-03-12 14:00:00"
}
```

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -119,3 +127,29 @@ message DeleteRemoteRequest{
...
+
+// Request to list all partitions from a named database
+message ListPartitionsRequest {
+  // the name of the database
+  string db_name = 1;
+}
+
+message ListPartitionsResponse {
+  // All partition keys in a database
+  repeated string partition_keys = 1;
```
Comment (pauldix) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:139]: Maybe future proof this by making this a message of `Partition` and have it only have a single field called `key`?

**Review by pauldix — APPROVED** [review]
Just one comment about the proto definition. But this could go out as is and we can just make that breaking change later when we need it, which will come before anyone else is using this API.

File: generated_types/protos/influxdata/iox/management/v1/service.proto (hunk 1 above)
Comment (alamb) [reply generated_types/protos/influxdata/iox/management/v1/service.proto:139]: Good idea -- will do

Comment (alamb) [reply generated_types/protos/influxdata/iox/management/v1/service.proto]: fixed in https://github.com/influxdata/influxdb_iox/pull/983/commits/c632b2a8d6c1527cccc6725050dbbec82c066047

**Pull request merged**